### ISSUE TITLE
Trying to change server address (URL and/or hostname) fails

### ISSUE THREAD

**Issue by OlivierS1** [nextcloud/android#1258]
### Actual behaviour
-Tell us what happens
I am experimenting with a server located on my LAN. I had it setup not to support https so I was logging only with http://.
I now have setup a SSL certificate and I need to update the server address on my phone.
Upon reconnecting to the server NC Dev prompted me with a certificate request approval. I accepted it, but to make sure I got the proper "App Password" I decided to delete it from the server.
Because the "App Password" no longer exist, NC Dev is crashing, but then I am able to regain control of NC Dev,
Trying to change the password, I am presented with the server address which is based on http://.
I cannot edit the address and change http:// into https://.
The only way to go by is to delete the account and start fresh.

One nasty side effect, is that unless Data and Wifi are disabled, NC Dev will force a switch to the server configuration page, which prevent user account switch, deletion or creation.

### Expected behaviour
-Tell us what should happen
I should be able to edit the server address to make proper correction.
Alos, the switch to edit server correction should not be forced upon. Instead it should allow for account managemenrt capability.

### Steps to reproduce
1. 
2. 
3. 

### Environment data
Android version:
6.0
Device model: 
LG 3
Stock or customized system:
Stock
Nextcloud app version:
NC Dev 20170802
Nextcloud server version:
NC 12
#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by tobiasKaminsky** [comment]
If I remember correctly we agreed that changing the server url should not be possible due to security concerns: an user could take your phone and just change the url without notice...

**Comment by OlivierS1** [comment]
@tobiasKaminsky I can accept this scenario if it to improve security.
Then there must be a way to inform the user that the address cannot be modified.
I was hopping during my migration that after revoking the "app password" from the server I would be asked to login through  the server page again, but becasue of the http:/ unsecure, NC Dev couldn't establish connection to server.
I couldn't enter a new "app password" not even being able to delete the account. My only viable solution was to uninstall NC Dev from my phone.
This leave the following question behind:
Upon reinstalling, the original user account which couldn't be removed may still contain synchronized data on the phone, so upon installing the new account (especially if the account was saved on the external memory media), upon installing NC Dev, the default location for the account  is going to be the internal memory.
Upon moving the account to the exteernal memory memory media, the old account folder will be reported by NC Dev.
There are 2 options to select from:
1) Replace old data
2) use existing data

With option 1), what will happen to files that could not have been shared, but moved or created under this account. Files that could have been made available offline could have been modified locally and the content would have been lost. It is possible Autoupload or other applications could have copied files into user account folder but not been synchronized yet.
So in a nutshell, there could be relative important loss of data.

With option 2) I don't know what really is going to happen. How will NC synchronize the data, will it create duplicates on server, will it modify the content of the files?

What c=kind of enhancement can be made to facilitate answering those questions?

Thinking of the security concern, nothing can prevent someone from using the phone and instead of switching the server address can potentially setup a new account and perform "Autoupload" from the legitimate account to the new server.
This is a remote possibility but a possibility nonetheless.

**Comment by j-ed** [comment]
If it is not allowed to change the personal settings due to security concerns, I would recommend to add a comment to the lower part of the screen (which is currently empty) stating that the displayed information are read-only settings. Otherwise users will ask this question again and again. Example mockup:

![screenshot_20170901-153032](https://user-images.githubusercontent.com/2572764/29972439-66d38c76-8f2c-11e7-9194-f1fcae9c15cc.png)

**Comment by paskalito** [comment]
actually i have the same problem..

my dyndns provider died so i NEED a enterely new adress. i would really hate to resync my dozens of gbytes of data just because of that ( on my slow inet conection.. and phone..).

i don't quiet get the security concerns. something like this is way easier done by hijacking the DNS. - and if you really worry maybe passwortprotecting these changes by the user passwort would be easier.

or even loads less easy just deleting the local password on server change and making it not possible to finish the change until you login again. this would solve the authentification issue..

**Comment by joshtrichards** [comment]
Duplicate of #4157 

Let's consolidate this enhancement idea at #4157 where most of the discussion has been occurring.
Subscribe there to track progress (if desired) and/or give that issue vote for prioritization.
